html/includes/api_functions.inc.php

```diff
@@ -1327,6 +1376,59 @@ function get_devices_by_group()
     api_success($devices, 'devices');
 }
 
+
+function list_vrf()
+{
+    $app        = \Slim\Slim::getInstance();
+    $sql        = '';
+    $sql_params = array();
+    $hostname   = $_GET['hostname'];
+    $vrfname    = $_GET['vrfname'];
+    $device_id  = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+    if (is_numeric($device_id)) {
+        $sql        = ' AND `vrfs`.`device_id`=?';
+        $sql_params = array($device_id);
+    }
+    if (!empty($vrfname)) {
+        $sql        = ' AND `vrfs`.`vrf_name`=?';
+        $sql_params = array($vrfname);
+    }
+    check_device_permission($device_id);
```

Comment:
This needs to be moved into the is_numeric check above otherwise we don't have a device_id potentially. Should fall back to is_read() if `if (is_numeric($device_id)) {` is false.

Reply:
Will fix in followup.